### PR TITLE
Add core code

### PR DESCRIPTION
**Pull request by alejandromumo**
- Base repo structure updated:
  - Added `black` formatter, `isort` and `pydocstyle`
  - Added `pytest` and  `tests/` folder
  - Added `db` class to manage db (session and engine)
  - Added base `Logger` class
  - Added custom errors to start with. 
- Added `init` command
  - Creates the a `lycophron.cfg` file with required configurations to run a project.
  - Creates a local database.
- Added `load` command
  - Reads a `.csv` file and processes it using `marshmallow`.
  - Dumps the read data in to the local database.
- Added `publish` command
  - Reads records from the local database and published to Zenodo.
  - Implemented using `celery` in multiple steps:
    - Spawns a base celery task that will run all the steps.
    - After creating a deposition, every file of a record is uploaded in parallel using a [group](https://docs.celeryq.dev/en/stable/reference/celery.html#celery.group).
    - When files group task is ready, a new task is linked to publish the record. See [celery docs](https://docs.celeryq.dev/en/stable/userguide/canvas.html#group-callbacks-and-error-handling) for more details on this approach.

Example of a record uploaded with lycophron: https://sandbox.zenodo.org/record/1159642

Notes:
- Only records with `DOI` are accepted for now. 
- Empty values on a record's field (e.g. keywords) are stripped and considered to be non-existent.

### CODE REVIEW

**Review by slint — APPROVED** [review]
Some minor comments (all potentially shelvable actually), otherwise looks great :)

File: src/lycophron/project.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        data = self.process_file(filename)
+        for record in data:
+            self.add_record(record)
+        if len(self.errors):
+            ErrorHandler.handle_error(self.errors)
+
+    def process_file(self, filename):
+        if not self.is_project_initialized():
+            raise Exception("Project is not initialised.")
+        factory = LoaderFactory()
```
Comment (alejandromumo) [inline src/lycophron/project.py:36]: using loader factory, tested and the behavior is the same as before. However, if the file format is not supported an exception is raised. Probably we want to expose that message to the user (e.g. "`.xslx` is not supported yet")

File: src/lycophron/tasks/tasks.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+"""Lycophron tasks implementation."""
+
+from celery import group
+import os
+
+from . import app
+from ..db import db
+from ..client import create_session
+
+# TODO tasks are using 'db' directly. Breaks the basic flow Interface -> Business -> Data.
```
Comment (alejandromumo) [inline src/lycophron/tasks/tasks.py:16]: It troubles me that tasks have a direct depedency to `db`, doesn't feel right. 

Is this ok?

File: src/lycophron/tasks/tasks.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+# TODO tasks do not have any logging implemented yet (e.g. success, errors).
+# TODO tasks are not updating the record's local status (e.g. 'SUCCESS', 'FAILED', etc)
```
Comment (alejandromumo) [inline src/lycophron/tasks/tasks.py:18]: This is important given that we do not have any means to assess whether a deposit was already created or not. We could be creating duplicates.

File: src/lycophron/__main__.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import sys
+from .cli import lycophron
+
+
+def init_logging():
+    """Initialises logging in two steps:
+    1 - adds a "user-friendly" logger that prints errors to stdout.
+    2 - adds a "dev-only" logger tha prints errors to a file.
+    """
+    logger = logging.getLogger("lycophron")
```
Comment (alejandromumo) [inline src/lycophron/__main__.py:13]: In general, logging could be much cleaner and modular.

File: src/lycophron/models.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# Lycophron is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Lycophron data models."""
+
+from sqlalchemy import Column, Integer, String, JSON, create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy_utils.models import Timestamp
+
+engine = create_engine("sqlite:///" + "data.db", echo=True)
```
Comment (slint) [inline src/lycophron/models.py:14]: shelve: we should take this from config

**Review by slint — APPROVED** [review]

File: src/lycophron/models.py (hunk 5 above)
Comment (alejandromumo) [reply src/lycophron/models.py:14]: I think this can be removed, no? The database is created in `db.py` using `config['SQLALCHEMY_DATABASE_URI']`

**Pull request merged**